src/core/components/ChatHistory.tsx

```diff
@@ -5,73 +5,27 @@ import {
   useEffect,
   useState,
   useMemo,
+  memo,
   Dispatch,
   SetStateAction,
-  memo,
-  ChangeEvent,
 } from 'react';
 import { useAppContext } from '../context/useAppContext';
-import NewChatIcon from '../assets/NewChatIcon';
 import { useIsMobile } from '../../shared/theme/useIsMobile';
 import { TrashIcon } from '../assets/TrashIcon';
 import KavaAILogo from '../assets/KavaAILogo';
 import { Pencil } from 'lucide-react';
+import { PenSquare } from 'lucide-react';
+import SearchModal from './SearchModal';
+import {
+  formatConversationTitle,
+  groupConversationsByTime,
+} from '../utils/conversation/helpers';
 
 interface ChatHistoryProps {
   setChatHistoryOpen: Dispatch<SetStateAction<boolean>>;
 }
 
-type GroupedConversations = {
-  [key: string]: ConversationHistory[];
-};
-
-const getTimeGroup = (timestamp: number): string => {
-  const now = new Date();
-  const diffDays = Math.floor(
-    (now.getTime() - timestamp) / (1000 * 60 * 60 * 24),
-  );
-
-  if (diffDays === 0) {
-    return 'Today';
-  } else if (diffDays === 1) {
-    return 'Yesterday';
-  } else if (diffDays <= 7) {
-    return 'Last week';
-  } else if (diffDays <= 14) {
-    return '2 weeks ago';
-  } else if (diffDays <= 21) {
-    return '3 weeks ago';
-  } else if (diffDays <= 28) {
-    return '4 weeks ago';
-  } else if (diffDays <= 60) {
-    return 'Last month';
-  } else if (diffDays <= 90) {
-    return '2 months ago';
-  } else {
-    const months = Math.floor(diffDays / 30);
-    return `${months} months ago`;
-  }
-};
-
-const groupConversations = (
-  conversations: ConversationHistory[],
-): GroupedConversations => {
-  const sortedConversations = [...conversations].sort(
-    (a, b) => b.lastSaved - a.lastSaved,
-  );
-
-  return sortedConversations.reduce((groups, conversation) => {
-    const timeGroup = getTimeGroup(conversation.lastSaved);
-    if (!groups[timeGroup]) {
-      groups[timeGroup] = [];
-    }
-    groups[timeGroup].push(conversation);
-    return groups;
-  }, {} as GroupedConversations);
-};
-
```

Comment:
Since this functionality now exists in the sidebar and in the search modal, I moved the functions to a helpers file and added unit tests.